client/signup/steps/business-details/index.tsx

```diff
@@ -0,0 +1,233 @@
+
+interface State {
+	suggestionsRequest: SuggestionsRequest;
+}
+
+class BusinessDetails extends Component< Props & ConnectedProps & LocalizeProps, State > {
+	state: State = {
+		suggestionsRequest: {
+			status: RequestStatus.NotRequested,
+		},
+	};
+
+	componentDidMount() {
+		this.props.saveSignupStep( { stepName: this.props.stepName } );
+	}
+
+	handleInputChange = ( action: 'setBusinessName' | 'setBusinessAddress' ) => ( {
+		currentTarget: { value = '' },
+	} ) => this.props[ action ]( value );
+
+	handleSubmit = ( placeId: string | null ) => {
+		const { businessName, flowName, stepName } = this.props;
+
+		// siteTitle is a required dependency to complete the signup flow.
+		this.props.submitSignupStep( { stepName, flowName }, { businessName, placeId } );
+		this.props.goToNextStep();
+	};
+
+	suggestionsSearch = ( event: React.FormEvent ) => {
+		event.preventDefault();
+		const { businessAddress, businessName } = this.props;
+		this.setState( { suggestionsRequest: { status: RequestStatus.Loading } } );
+
+		wpcom
+			.undocumented()
+			.getRivetSuggestions( { name: businessName, address: businessAddress } )
+			.then( ( response: any ) => {
+				this.setState( {
+					suggestionsRequest: {
+						status: RequestStatus.Success,
+						suggestions: get( response, 'suggestions[0].list', [] ),
+					},
+				} );
+			} )
+			.catch( ( response: any ) => {
+				this.setState( {
+					suggestionsRequest: {
+						status: RequestStatus.Error,
+						errorMessage: response.message || '',
+					},
+				} );
+			} );
+	};
+
+	renderSuggestions = () => {
+		const { translate } = this.props;
+		const { suggestionsRequest } = this.state;
+
+		switch ( suggestionsRequest.status ) {
+			case RequestStatus.Success:
+				return (
+					<Card>
+						{ suggestionsRequest.suggestions.length
+							? suggestionsRequest.suggestions.map( suggestion => (
+									<Card
+										displayAsLink
+										key={ suggestion.id }
+										onClick={ this.handleSubmit.bind( this, suggestion.id ) }
+									>
+										<div>
+											<strong>{ suggestion.title }</strong>
+										</div>
+										<div>{ suggestion.address }</div>
+									</Card>
+							  ) )
+							: translate( 'No results for that location.' ) }
+					</Card>
+				);
+			case RequestStatus.Error:
+				return (
+					<Card>
+						{ suggestionsRequest.errorMessage ||
+							translate( 'There was an error searching that location. Please try again.' ) }
+					</Card>
+				);
+			case RequestStatus.NotRequested:
+			case RequestStatus.Loading:
+				return null;
+		}
+	};
+
+	renderContent = () => {
+		const { businessAddress, businessName } = this.props;
+		const { suggestionsRequest } = this.state;
+
+		return (
+			<div className="business-details__wrapper">
+				<Card>
+					<form>
+						<FormFieldset className="business-details__field-control">
+							<FormLabel htmlFor="title">
+								{ this.props.translate( 'Your business name' ) }
```

Comment:
Hi! I've found a possible matching string that has already been translated **25** times:
`translate( 'Enter your business name' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).